### PR TITLE
Update multiple http to https links

### PR DESCRIPTION
**Pull request by kmpoppe**
Changed a bunch of http to https links, as already proposed by @einsteinsfool in #3134, yet the PR got a whole lot of conflicts, which is why we agreed to close it.

Also, where possible, URLs where shortened (when `.../<someid>/some-google-optimzed-text` yielded the same page as `.../<someid>`).

### CODE REVIEW

File: _data/banking.yml [hunk 1]
```diff
@@ -578,7 +578,7 @@ websites:
       img: intesasanpaolo.png
       tfa: Yes
       hardware: Yes
-      doc: http://www.intesasanpaolo.com/servizi-multicanale/sicurezza.jsp
+      doc: https://www.intesasanpaolo.com/servizi-multicanale/sicurezza.jsp
```
Comment (kmpoppe) [inline _data/banking.yml]:
```suggestion
      doc: https://www.intesasanpaolo.com/it/common/footer/sicurezza.html
```

**Review by RichJeanes — DISMISSED** [review]

**Review by RichJeanes — DISMISSED** [review]

File: _data/social.yml [hunk 2]
```diff
@@ -261,7 +261,7 @@ websites:
       doc: https://vk.com/page-59800369_47885415
 
     - name: Weibo
-      url: http://www.weibo.com
+      url: https://www.weibo.com/
```
Comment (Carlgo11) [inline _data/social.yml:264]: I'm unsure if they provide HTTPS on their regular site or only the overseas version. 🤔

Comment (Carlgo11) [reply _data/social.yml:264]: Checked on VPN from Hong Kong and they provided the site over HTTPS however the mainland could have a different setup. Also, it's impossible to change language as that option is in the footer and their JS-script autoloads more content when the screen reaches the footer 😠

File: _data/utilities.yml [hunk 3]
```diff
@@ -450,7 +450,7 @@ websites:
       tfa: No
 
     - name: Vodafone (IT)
-      url: http://www.vodafone.it/
+      url: https://www.vodafone.it/
```
Comment (Carlgo11) [inline _data/utilities.yml]: I can't load the site over HTTPS. Is it just me?
Maybe they don't support modern ciphers or something. 🤔

File: _data/social.yml (hunk 2 above)
Comment (kmpoppe) [reply _data/social.yml:264]: https:// is the direct responseUri I get from the HTTP-request, the forward to /overseas is in a later step, so there must be some form of https involved.

File: _data/utilities.yml (hunk 3 above)
Comment (kmpoppe) [reply _data/utilities.yml]: Firefox, doing TLS 1.2, redirects to https://www.vodafone.it/portal/Privati immediately when entering the https URL, I wouldn't know where they get the information from, that it's a private account I'd want to open, but https works for me.

Comment (Carlgo11) [reply _data/utilities.yml]: Found the issue:

chrome_2019-05-07_21-19-23
firefox_2019-05-07_21-20-13
firefox_2019-05-07_21-19-48

Maybe we should continue with the HTTP version for now? I'm guessing more users than me are going to have problems loading the site over HTTPS.

**Review by Carlgo11 — CHANGES_REQUESTED** [review]

File: _data/utilities.yml (hunk 3 above)
Comment (kmpoppe) [reply _data/utilities.yml]: Ah, ffs, Vodafone ...

Comment (kmpoppe) [inline _data/utilities.yml]:
```suggestion
      url: http://www.vodafone.it/
```

**Review by Carlgo11 — APPROVED** [review]

**Review by RichJeanes — APPROVED** [review]

**Pull request merged**